File: design-patterns/go/composite/player_group/cricket_player.go
// cricket_player.go

package main

import "fmt"

type CricketPlayer struct {
	name string
	age  int
	run  int
}

func NewCricketPlayer(name string, age int, run int) (cricketPlayer *CricketPlayer) {
	cricketPlayer = &CricketPlayer{}
	cricketPlayer.name = name
	cricketPlayer.age = age
	cricketPlayer.run = run
	return
}

func (cricketPlayer *CricketPlayer) PrintDetails() {
	fmt.Println("Game: Cricket")
	fmt.Printf("Name: %v\n", cricketPlayer.name)
	fmt.Printf("Age: %v\n", cricketPlayer.age)
	fmt.Printf("Runs: %v\n", cricketPlayer.run)
}

File: design-patterns/php/composite/playergroup/CricketPlayer.php
<?php
// CricketPlayer.php

namespace BigBoxCode\DesignPattern\Composite\PlayerGroup;


class CricketPlayer implements Player {
    public function __construct(
        private string $name,
        private int $age,
        private int $run
    ) {
       
    }

    public function printDetails(): void {
        echo "\nGame: Cricket";
        echo "\nName: " . $this->name;
        echo "\nAge: " . $this->age;
        echo "\nRuns: " . $this->run;
    }
}